### PR TITLE
fix: chat issues

### PR DESCRIPTION
**Pull request by jitendraharpalani**
https://bloomstack.com/desk#Form/Task/TASK-2020-01520
https://bloomstack.com/desk#Form/Task/TASK-2020-00259
https://bloomstack.com/desk#Form/Task/TASK-2020-01521
https://bloomstack.com/desk#Form/Task/TASK-2020-01522

Summary
https://bloomstack.com/desk#Form/Task/TASK-2020-01097

### CODE REVIEW

File: frappe/public/js/frappe/chat.js [hunk 1]
```diff
@@ -2152,23 +2229,38 @@ class extends Component {
...
+						console.log("fire seen !");
 						frappe.chat.message.seen(message.name)
-					else
+					} else {
 						break
+					}
+				}
 			}
+
+			 */
```
Comment (hrwX) [inline frappe/public/js/frappe/chat.js:2260]: - If this is not required then lets remove it ?

File: frappe/public/js/frappe/chat.js [hunk 2]
```diff
@@ -2680,10 +2787,13 @@ frappe.chat.render = (render = true, force = false) =>
 		// Render if frappe-chat-toggle doesn't exist.
 		if ( frappe.utils.is_empty($placeholder.has('.frappe-chat-toggle')) ) {
 			const $template = $(`
-				<a class="dropdown-toggle frappe-chat-toggle" data-toggle="dropdown">
+				<a class="dropdown-toggle frappe-chat-toggle" data-toggle="dropdown" id="chat_toggle_navbar">
 					<div>
 						<i class="octicon octicon-comment-discussion"/>
 					</div>
+					<span class="notifications-indicator" style="display: none; color: red; font-size:10px; top: 0; right: 5px" id="unread_message_indicator">
```
Comment (hrwX) [inline frappe/public/js/frappe/chat.js]: - We can show a similar indicator just like the one notification icon has right ??
- The red color is kinda looking out of place.

File: frappe/public/js/frappe/chat.js [hunk 3]
```diff
@@ -2000,19 +2073,20 @@ class extends Component {
...
 			h("li", null,
 				h("a", { class: props.active ? "active": "", onclick: () => {
-					if (props.last_message) {
-						frappe.chat.message.seen(props.last_message.name);
-					}
-					props.click(props)
+					props.click(props);
+					// if (props.last_message) {
+					// 	frappe.chat.message.seen(props.last_message.name);
+					// }
```
Comment (hrwX) [inline frappe/public/js/frappe/chat.js]: - Remove this too

File: frappe/public/js/frappe/chat.js [hunk 4]
```diff
@@ -2193,6 +2286,16 @@ class extends Component {
...
+						h("button", {
+								class: "dropdown-toggle frappe-chat-toggle",
+								style: { "font-size": "9px" },
+								onclick: () => {
+									var expandButton = document.getElementById("resize_button");
+									expandButton.click();
+									var chatButtonToggle = document.getElementById("chat_toggle_navbar");
+									chatButtonToggle.click();
+								}
+								}, "X"),
```
Comment (hrwX) [inline frappe/public/js/frappe/chat.js]: can we add an octicon here ?? the `X` looks bad

image

**Review by hrwX — CHANGES_REQUESTED** [review]

File: frappe/public/js/frappe/chat.js (hunk 1 above)
Comment (vigilantee) [reply frappe/public/js/frappe/chat.js:2260]: We will be replacing the chat module in the near so we might need this for reference.

**Review by vigilantee — APPROVED** [review]

**Comment by hrwX** [comment]
Closing this PR since a version of this has been merged in develop

**Pull request closed without merging**